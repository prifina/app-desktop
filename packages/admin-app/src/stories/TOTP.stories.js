import React from "react";

import * as OTPAuth from "otpauth";
import CryptoJS from 'crypto-js'

import { Auth } from "aws-amplify";

import config from "../config";
//import { TOTP } from '@lgicc/totp';

let AUTHConfig = {
  // To get the aws credentials, you need to configure
  // the Auth module with your Cognito Federated Identity Pool
  mandatorySignIn: false,
  userPoolId: config.cognito.USER_POOL_ID,
  identityPoolId: config.cognito.IDENTITY_POOL_ID,
  userPoolWebClientId: config.cognito.APP_CLIENT_ID,
  region: config.auth_region,
  identityPoolRegion: config.main_region,
  //region: config.main_region,
  //authenticationFlowType: 'CUSTOM_AUTH'
};
export default { title: "TOTP" };

function randomString() {
  return Math.random()
    .toString(36)
    .slice(2).toUpperCase();
}

const generateSecretASCII = (length = 32, symbols = true) => {
  //const bytes = crypto.randomBytes(length || 32)
  const wordArray = CryptoJS.lib.WordArray.random(length)

  const bytes = CryptoJS.enc.Base64.stringify(wordArray);
  //.toString(CryptoJS.enc.Utf8);
  console.log("BYTES ", bytes)
  let set = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXTZabcdefghiklmnopqrstuvwxyz"
  if (symbols) {
    set += "!@#$%^&*()<>?/[]{},.:;"
  }

  let output = ""
  for (let i = 0, l = bytes.length; i < l; i++) {
    output += set[Math.floor((bytes[i] / 255.0) * (set.length - 1))]
  }
  return output

}


const totpVerifyDelta = (options) => {
  // shadow options
  options = Object.create(options);
  // verify secret and token exist
  var secret = options.secret;
  var token = options.token;
  if (secret === null || typeof secret === 'undefined') throw new Error('Speakeasy - totp.verifyDelta - Missing secret');
  if (token === null || typeof token === 'undefined') throw new Error('Speakeasy - totp.verifyDelta - Missing token');

  // unpack options
  var window = parseInt(options.window, 10) || 0;

  // calculate default counter value
  if (options.counter == null) options.counter = exports._counter(options);

  // adjust for two-sided window
  options.counter -= window;
  options.window += window;

  // pass to hotp.verifyDelta
  var delta = exports.hotp.verifyDelta(options);

  // adjust for two-sided window
  if (delta) {
    delta.delta -= window;
  }

  return delta;
};


export const totp = () => {
  //console.log("SECRET ", generateSecretASCII());
  //console.log("SECRET ", randomString());
  //otpauth://totp/ACME%20Co:john.doe@email.com?secret=HXDMVJECJJWSRB3HWIZR4IFUGFTMXBOZ&issuer=ACME%20Co&algorithm=SHA1&digits=6&period=30
  //const key = randomString();
  const key = "TUMQQC3PJJ";
  console.log("SECRET ", key);
  //  TUMQQC3PJJ

  const companyName = "Prifina";
  const userAccount = "xxxx-zzzz-wwww";
  const otpauth = 'otpauth://totp/' + encodeURI(companyName) + ':' + encodeURI(userAccount) + "?issuer=" + encodeURI(companyName) + '&secret=' + key.replace(/\s+/g, '').toUpperCase() + "&algorithm=SHA1&digits=6&period=30";

  let totp = new OTPAuth.TOTP({
    issuer: companyName,
    label: userAccount,
    algorithm: "SHA1",
    digits: 6,
    period: 30,
    secret: key, // or 'OTPAuth.Secret.fromBase32("NB2W45DFOIZA")'
  });


  // Generate a token.
  let token = totp.generate();
  console.log("TOKEN ", token);
  // Validate a token.
  let delta = totp.validate({
    token: token,
    window: 1,
  });

  console.log("DELTA ", delta);

  /*
  let token = "855 714";
  console.log("TOKEN ", token);
  // Validate a token.
  let delta = totp.validate({
    token: token,
    window: 1,
  });
  
  console.log("DELTA ", delta);
  */
  // Convert to Google Authenticator key URI:
  //   otpauth://totp/ACME:AzureDiamond?issuer=ACME&secret=NB2W45DFOIZA&algorithm=SHA1&digits=6&period=30
  let uri = totp.toString();
  console.log("URI ", uri, otpauth);
  let parsedTotp = OTPAuth.URI.parse(uri);
  console.log("PARSED ", parsedTotp);
  // const src = 'https://chart.googleapis.com/chart?chs=166x166&chld=L|0&cht=qr&chl=' + encodeURIComponent(uri);
  const src = 'https://chart.googleapis.com/chart?chs=166x166&chld=L|0&cht=qr&chl=' + encodeURIComponent(otpauth);

  Auth.configure(AUTHConfig);
  let user = undefined;
  return <>
    <div>Testing</div>
    <img src={src} />
    <div><input id="token" defaultValue={"123456"} /></div>
    <button onClick={async () => {
      const token = document.getElementById("token").value;
      let delta = totp.validate({
        token: token,
        window: 1,
      });

      console.log("DELTA ", delta);
      const result = await verify({ token, secret: key });
      console.log(key, result);

    }}>CHECK</button>
    <button onClick={async () => {
      user = await Auth.signIn("test-user");
      console.log("LOGIN", user);
      //const result = await generateToken({ secret: key });
      //console.log(result)


    }}>SIGN IN CUSTOM</button>
    <button onClick={async () => {
      user = await Auth.signIn("test-user", "");
      console.log("LOGIN", user);
      //const result = await generateToken({ secret: key });
      //console.log(result)


    }}>SIGN IN DEFAULT</button>
    <button onClick={async () => {
      //let user = await Auth.signIn("test-user");
      const token = document.getElementById("token").value;
      const result = await Auth.sendCustomChallengeAnswer(user, token);
      //const result = await generateToken({ secret: key });
      console.log(result)


    }}>VALIDATE </button>

  </>
};

totp.storyName = "TOTP";

/*
  var otpauth = 'otpauth://totp/'
    + encodeURI(companyName) + ':' + encodeURI(userAccount)
    + '?secret=' + key.replace(/\s+/g, '').toUpperCase()
    ;
  */
//var src = 'https://chart.googleapis.com/chart?chs=166x166&chld=L|0&cht=qr&chl=' + encodeURIComponent(otpauth);


/*
Authenticator.verifyToken(key, token).then(function (result) {
  var msg;
  if (result) {
    msg = 'Correct!';
  } else {
    msg = 'FAIL!';
  }

  console.info('verify', msg);
  window.alert(msg);
}, function (err) {
  window.alert('[ERROR]:' + err.message);
  window.alert('[ERROR]:' + err.stack);

  console.error('ERROR');
  console.error(err);
});
});

*/