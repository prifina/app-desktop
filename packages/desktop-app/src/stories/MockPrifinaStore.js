import React from "react";
import { Provider, } from "../stores/PrifinaStore";

import create from 'zustand'

export const MockStore = ({ checkEmail, checkUser, checkPhone, verifyResult = true, children }) => {
  console.log("MOCK STORE ", checkEmail, checkUser, checkPhone);
  const emailCheckResult = checkEmail || false;
  const userCheckResult = checkUser || false;
  const phoneCheckResult = checkPhone || false;
  return <Provider createStore={() =>
    create((set, get) => ({
      activeIndex: -1,
      setActiveIndex: (newIndex) => {
        set({ activeIndex: newIndex });
      },
      signUp: async (newUser) => {
        return Promise.resolve({ user: newUser });
      },
      getCountryCodeQuery: async () => {
        return Promise.resolve("FI");
      },
      checkCognitoAttributeQuery: async (attrName, attrValue) => {
        /*
        API,
        "email",
        email,
        config.cognito.USER_POOL_ID,
        */
        if (attrName === "email") return Promise.resolve(emailCheckResult);
        if (attrName === "username") return Promise.resolve(userCheckResult);
        if (attrName === "phone_number") return Promise.resolve(phoneCheckResult);
      },
      resendCode: async (target) => {
        return Promise.resolve(true);
        /*
      try {
        await sendVerificationMutation(
          API,
          "email",
          JSON.stringify({
            userId: currentUser.username,
            clientId: currentUser.client,
            email: currentUser.email,
            given_name: currentUser.given_name,
          }),
        );
        alerts.info(i18n.__("emailVerificatioSent"), {});
      } catch (e) {
        console.log("ERR", e);
      }
      */
      },
      verifyCode: async (code) => {
        return Promise.resolve(verifyResult);
        /*
          try {
            const userCode = [
              currentUser.username,
              currentUser.client,
              "email",
              verificationFields.verificationCode,
            ].join("#");
      
            const result = await getVerificationQuery(API, userCode);
      
            if (result.data.getVerification === null) {
              alerts.error(i18n.__("invalidCode"), {});
              setInputError({ status: true });
            }
            console.log("VERIFY ", result);
            nextStepAction(3);
          } catch (e) {
            console.log("ERR", e);
            alerts.error(i18n.__("invalidCode"), {});
            setInputError({ status: true });
            nextStepAction(2);
          }
          */
      }


    }))
  }>
    {children}
  </Provider>
}
