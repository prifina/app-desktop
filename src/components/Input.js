import React, { forwardRef } from "react";

import styled, { css } from "styled-components";
import {
  space,
  layout,
  typography,
  border,
  color,
  compose,
} from "styled-system";
//import PropTypes from "prop-types";

const systemProps = compose(layout, color, space, border, typography);

const inputTheme = css`
  appearance: none;
  display: block;
  font-family: inherit;
  font-size: ${props => props.theme.componentStyles[props.as].base.fontSize};
  line-height: ${props =>
    props.theme.componentStyles[props.as].base.lineHeight};
  color: inherit;
  background-color: transparent;
  border: ${props =>
    typeof props.borders !== "undefined"
      ? props.borders
      : props.theme.componentStyles[props.as].base.border};
  border-radius: ${props =>
    typeof props.borderRadius !== "undefined"
      ? props.borderRadius
      : props.theme.componentStyles[props.as].base.borderRadius};
  margin: 0;
  padding-left: ${props =>
    props.theme.componentStyles[props.as].base.paddingLeft};
  padding-right: ${props =>
    props.theme.componentStyles[props.as].base.paddingRight};
  padding-top: ${props =>
    props.theme.componentStyles[props.as].base.paddingTop};
  padding-bottom: ${props =>
    props.theme.componentStyles[props.as].base.paddingBottom};
  ::placeholder {
    color: ${props => props.theme.colors.text.muted};
  }
  &:disabled {
    opacity: 0.25;
  }
  &:focus {
    outline: none;
    -webkit-box-shadow: none;
    box-shadow: none;
  }
  ::-ms-clear {
    display: none;
  }
  ${systemProps}
`;
const themeColorStyles = props => {
  // console.log("custom ", props);
  return props.colorStyle ? props.theme.colorStyles[props.colorStyle] : null;
};
const InputElement = styled.input`
  ${inputTheme}
  ${themeColorStyles}
`;

const Input = forwardRef((props, ref) => {
  // console.log("INPUT  ", props, ref);
  return <InputElement {...props} ref={ref} />;
});
Input.defaultProps = {
  as: "input",
  type: "text",
  width: "100%",
};

Input.displayName = "Input";
Input.isField = true;
/*
Input.propTypes = {
  type: PropTypes.string.isRequired,
};
*/
export default Input;
