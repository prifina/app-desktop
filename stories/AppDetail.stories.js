import React from "react";
import AppDetail from "../src/pages/AppDetail";

export default { title: "AppDetail" };

export const container = () => (
  <AppDetail progress={33} />
);
container.story = {
  name: "AppDetail",
};