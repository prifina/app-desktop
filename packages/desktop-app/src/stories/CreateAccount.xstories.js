/* eslint-disable react/prop-types */
import React, { useEffect, useRef, useState } from "react";

import { Routes, Route, Outlet, useNavigate, useLocation, useParams, } from "react-router-dom";


import { MockStore } from "./MockPrifinaStore";

import { useStore } from "../stores/PrifinaStore";

import CreateAccount from "../pages/CreateAccount-v2";

import { Router } from "./utils";

import { ToastContextProvider, useToast } from "@blend-ui/toast";

const PageLogin = () => {
  const activeIndex = useStore(state => state.activeIndex);
  console.log("ROUTE PAGE LOGIN ", activeIndex);
  return <><div>Page Login
  </div>
  </>
}

const PageA = () => {
  const activeIndex = useStore(state => state.activeIndex);
  console.log("ROUTE PAGE A ", activeIndex);
  return <><div>Page A
  </div>
    {/* 
    <Outlet />
    */}
  </>
}



const PageAI = () => "Page AI";
const PageAWithHash = () => "Page A with Hash";
const PageB = () => "Page B";
const PageC = () => "Page C";


const Routing = () => {
  const navigate = useNavigate();
  const { pathname } = useLocation();
  const { id } = useParams();
  console.log("ROUTE ", pathname, id);
  const setActiveIndex = useStore(state => state.setActiveIndex);
  //console.log("HIST ", window.history)
  return <><Routes>
    <Route path="/test" element={<PageA />} />
    <Route path="/login" element={<PageLogin />} />

    <Route path="/register/*" element={<CreateAccount />} />

  </Routes>
    <button onClick={() => {
      setActiveIndex(0);
      navigate("/register", { replace: true });
    }}>REGISTER</button>
    <button onClick={() => {
      setActiveIndex(1);
      navigate("/register/terms-of-use", { replace: true });
    }}>PAGE B</button>
    <button onClick={() => {
      setActiveIndex(2);
      navigate("/register/email-verification", { replace: true });
    }}>PAGE C</button>
    <button onClick={() => {
      setActiveIndex(2);
      navigate("/register/phone-verification", { replace: true });
    }}>PAGE D</button>
    <button onClick={() => {
      setActiveIndex(1);
      navigate("./");
      //navigate("./");
    }}>BACK1</button>
    <button onClick={() => {
      setActiveIndex(0);
      navigate("../");
    }}>BACK2</button>
  </>

}

export default {
  title: "Pages /CreateAccount",
  component: Routing,

};



const Template = args => <Routing {...args} />;
export const CreateAccountSB = Template.bind({});

CreateAccountSB.storyName = "SB";


const activeMockupDecorator = (story, ctx) => {

  //console.log("CTX ",ctx.parameters)
  // can change verifyResult by using prop "verifyResult", default is true 
  return <><MockStore ><div>{story()}</div></MockStore></>
}


const toastProviderDecorator = (Story) => (
  <ToastContextProvider ><Story /></ToastContextProvider>
);

const routerDecorator = (story, ctx) => {
  return <><Router entries={["/register"]}>
    {story()}
  </Router></>
}


CreateAccountSB.decorators = [routerDecorator, activeMockupDecorator, toastProviderDecorator]
/*
const hashRouterProviderDecorator = (Story) => (
  <Routes>
    <Story />
  </Routes>
);


CreateAccountSB.decorators = [hashRouterProviderDecorator]
*/