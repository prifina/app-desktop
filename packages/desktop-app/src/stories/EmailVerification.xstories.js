/* eslint-disable react/prop-types */
import React, { useEffect, useRef, useState } from "react";
import { within, userEvent, fireEvent, waitFor, screen } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

import { action } from '@storybook/addon-actions';
//import { actions } from '@storybook/addon-actions';

import { ToastContextProvider, useToast } from "@blend-ui/toast";
import EmailVerification from "../pages/EmailVerification-v2";

import { useTranslate, } from "@prifina-apps/utils";
import { MockStore } from "./MockPrifinaStore";
import { useStore } from "../stores/PrifinaStore";
import shallow from 'zustand/shallow'
//import { rest } from 'msw'
//import { worker } from './mocks/browser'


const sleep = ms => new Promise(r => setTimeout(r, ms));

const supportLink = "/support";

export default {
  title: "Pages Sub/EmailVerification",
  component: EmailVerification,
  args: {
    invalidLink: supportLink
  },
  argTypes: {
    backClick: { action: "Back Click" },
    verifyClick: { action: "Verify Click" },
    resendClick: { action: "Resend Click" },
    //onClickSave: actionSave ,
  },

};


// This will lead to { onClick: action('onClick'), ... }
//const eventsFromNames = actions('onClick', 'onMouseOver');

// This will lead to { onClick: action('clicked'), ... }
//const eventsFromObject = actions({ onClick: 'clicked', onMouseOver: 'hovered' });



const Template = args => <EmailVerification {...args} />;
export const EmailVerificationSB = Template.bind({});

// this works, but is not stopping href navigation... 
//EmailVerificationSB.parameters = { actions: { handles: ['click .link'] } };

EmailVerificationSB.storyName = "SB";

const toastProviderDecorator = (Story) => (
  <ToastContextProvider ><Story /></ToastContextProvider>
);

/*
const actionSave = (data) => {
  console.log("SAVE ACTION ", data);
  //console.log("LOG" ,action("Update action"));
  // action return the handler which allows to update logging... 
  const actionHandler = action();
  actionHandler("save action", data)
  return Promise.resolve(true);
}
*/

const AddListener = () => {
  const [link, setLink] = useState("")
  const preventClick = (e) => {
    //console.log('A link was clicked');
    //console.log("LINK ", e.target.href);
    setLink(e.target.href);
    const actionHandler = action();
    actionHandler("Support link clicked");
    e.preventDefault();
  }
  useEffect(() => {

    const link = document.querySelector('a.link');
    const listener = link.addEventListener('click', preventClick);
    return () => {
      if (listener) {
        link.removeEventListener("click", preventClick);
      }
    }
  })
  return <div>{link}</div>
}

const verificationDecorator = (story, ctx) => {

  //console.log("CTX ",ctx.parameters)
  const { __ } = useTranslate();

  ctx.parameters["invalidTxt"] = __("codeDigitsError");

  return <><MockStore >{story()}</MockStore>
    <AddListener />
  </>
}

// Use Storybook decorators and MSW runtime handlers
// to handle the same HTTP call differently for this particular story.
/*
const linkDecorator = (story, ctx) => {
  worker.use(
    rest.get(supportLink, (req, res, ctx) => {

      return res(ctx.json({ message: 'OK' }))
    })
  )
  return <>{story()}</>
}
*/
/*
EmailVerificationSB.parameters = {
  msw: {
    handlers: [
      rest.get(supportLink, (req, res, ctx) => {
        return res(
          ctx.text("OK")
        )
      }),
    ]
  },
}
*/

EmailVerificationSB.decorators = [toastProviderDecorator, verificationDecorator]


export const EmailVerificationSBInteractive = Template.bind({});


const activeMockupDecorator = (story, ctx) => {

  //console.log("CTX ",ctx.parameters)
  const { __ } = useTranslate();
  ctx.parameters["invalidTxt"] = __("codeDigitsError");
  // can change verifyResult by using prop "verifyResult", default is true 
  return <><MockStore ><div>{story()}</div></MockStore></>
}


// copy args...
EmailVerificationSBInteractive.args = { ...EmailVerificationSB.args }

EmailVerificationSBInteractive.decorators = [toastProviderDecorator, verificationDecorator]

EmailVerificationSBInteractive.storyName = "Interactive";

const PlayTest = async ({ args, canvasElement, parameters }) => {

  console.log("ARGS ", args, parameters);

  const canvas = within(canvasElement);
  const input = canvas.getByTestId('verificationCode');
  console.log("INP ", input);
  const wrongCode = '1234a';
  const goodCode = '123456';

  await userEvent.click(canvas.getByRole("back-click"));
  await expect(args.backClick).toHaveBeenCalled();

  const verifyButton = canvas.getByRole("verify-click");

  // should be still disabled...
  expect(verifyButton.disabled).toBe(true)

  await userEvent.type(input, wrongCode);
  await waitFor(async () => {
    await expect(screen.getByText(parameters.invalidTxt)).toBeInTheDocument();
  });

  //await userEvent.click(canvas.getByRole("verify-click"));
  // should be disabled...
  //await expect(args.verifyClick).not.toHaveBeenCalled();

}


EmailVerificationSBInteractive.play = PlayTest;



// to mockup HOC-component which calls EmailVerification
const ActiveTemplate = args => {

  // this doesn't work in decorator level ... why toast works?
  const { resendCode, verifyCode } = useStore((state) => ({ verifyCode: state.verifyCode, resendCode: state.resendCode }),
    shallow
  );

  const { __ } = useTranslate();
  const alerts = useToast();

  //console.log("STORE", checkCognitoAttributeQuery);

  args.resendClick = () => {
    const actionHandler = action();
    actionHandler("Resend clicked", "email");
    resendCode("email").then(res => {
      alerts.info(__("emailVerificatioSent"), {});
    });
  }

  args.verifyClick = (code) => {
    const actionHandler = action();
    actionHandler("Verify clicked", code);
    verifyCode(code).then(res => {
      console.log("Verified...", res);
    })

  }
  //console.log("ACIVE MOCKUP", args);


  return <>
    <EmailVerification {...args} />
    <AddListener />
  </>
};

export const EmailVerificationSBActive = ActiveTemplate.bind({});


// copy args...
EmailVerificationSBActive.args = { ...EmailVerificationSB.args }

EmailVerificationSBActive.decorators = [toastProviderDecorator, activeMockupDecorator]

EmailVerificationSBActive.storyName = "Active";
