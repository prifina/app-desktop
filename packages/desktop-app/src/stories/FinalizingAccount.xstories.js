/* eslint-disable react/prop-types */
import React, { useEffect, useRef, useState } from "react";
import { within, userEvent, fireEvent, waitFor, screen } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

import { action } from '@storybook/addon-actions';
//import { actions } from '@storybook/addon-actions';

import { Routes, Route, } from "react-router-dom";


import { ToastContextProvider, useToast } from "@blend-ui/toast";
import FinalizingAccount from "../pages/FinalizingAccount-v2";

import { MockStore } from "./MockPrifinaStore";
import { useStore } from "../stores/PrifinaStore";
import shallow from 'zustand/shallow'

import { Router } from "./utils";

const newUser = {
  username: "xxxx-zzzz-vvvv-aaaa",
  password: "password",
  emailVerified: "troxxxx@gmail.com",
  phoneVerified: "+35812345678",
  lastName: "last-name",
  firstName: "first-name",
  preferred_username: "login-username"
};


const PageLogin = () => {

  const actionHandler = action();
  actionHandler("NAV LOGIN");

  return <><div>Page Login
  </div>
  </>
}


const Routing = () => {

  return <><Routes>

    <Route path="/login" element={<PageLogin />} />

    <Route path="/register/*" element={<FinalizingAccount currentUser={newUser} />} />

  </Routes>

  </>

}

export default {
  title: "Pages Sub/FinalizingAccount",
  component: Routing,

};



const Template = args => <Routing {...args} />;
export const FinalizingAccountSB = Template.bind({});


FinalizingAccountSB.storyName = "SB";

const toastProviderDecorator = (Story) => (
  <ToastContextProvider ><Story /></ToastContextProvider>
);


const activeMockupDecorator = (story, ctx) => {

  return <><MockStore ><div>{story()}</div></MockStore></>
}

const routerDecorator = (story, ctx) => {
  return <><Router entries={["/register"]}>
    {story()}
  </Router></>
}
FinalizingAccountSB.decorators = [toastProviderDecorator, activeMockupDecorator, routerDecorator]



const Template2 = args => <FinalizingAccount {...args} />;
export const FinalizingAccountNoRoutingSB = Template2.bind({});

FinalizingAccountNoRoutingSB.args = { currentUser: newUser };

FinalizingAccountNoRoutingSB.storyName = "SB-NoRouting";


FinalizingAccountNoRoutingSB.decorators = [toastProviderDecorator, activeMockupDecorator, routerDecorator]