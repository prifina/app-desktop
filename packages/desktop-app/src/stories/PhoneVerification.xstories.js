/* eslint-disable react/prop-types */
import React, { useEffect, useRef, useState } from "react";
import { within, userEvent, fireEvent, waitFor, screen } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

import { action } from '@storybook/addon-actions';
//import { actions } from '@storybook/addon-actions';

import { ToastContextProvider, useToast } from "@blend-ui/toast";
import PhoneVerification from "../pages/PhoneVerification-v2";

import { useTranslate, } from "@prifina-apps/utils";
import { MockStore } from "./MockPrifinaStore";
import { useStore } from "../stores/PrifinaStore";
import shallow from 'zustand/shallow'
//import { rest } from 'msw'
//import { worker } from './mocks/browser'


const sleep = ms => new Promise(r => setTimeout(r, ms));

const supportLink = "/support";

export default {
  title: "Pages Sub/PhoneVerification",
  component: PhoneVerification,
  args: {
    invalidLink: supportLink
  },
  argTypes: {
    backClick: { action: "Back Click" },
    verifyClick: { action: "Verify Click" },
    resendClick: { action: "Resend Click" },
    //onClickSave: actionSave ,
  },

};


// This will lead to { onClick: action('onClick'), ... }
//const eventsFromNames = actions('onClick', 'onMouseOver');

// This will lead to { onClick: action('clicked'), ... }
//const eventsFromObject = actions({ onClick: 'clicked', onMouseOver: 'hovered' });



const Template = args => <PhoneVerification {...args} />;
export const PhoneVerificationSB = Template.bind({});

// this works, but is not stopping href navigation... 
//EmailVerificationSB.parameters = { actions: { handles: ['click .link'] } };

PhoneVerificationSB.storyName = "SB";

const toastProviderDecorator = (Story) => (
  <ToastContextProvider ><Story /></ToastContextProvider>
);

const AddListener = () => {
  const [link, setLink] = useState("")
  const preventClick = (e) => {
    //console.log('A link was clicked');
    //console.log("LINK ", e.target.href);
    setLink(e.target.href);
    const actionHandler = action();
    actionHandler("Support link clicked");
    e.preventDefault();
  }
  useEffect(() => {

    const link = document.querySelector('a.link');
    const listener = link.addEventListener('click', preventClick);
    return () => {
      if (listener) {
        link.removeEventListener("click", preventClick);
      }
    }
  })
  return <div>{link}</div>
}

const verificationDecorator = (story, ctx) => {

  //console.log("CTX ",ctx.parameters)
  const { __ } = useTranslate();

  ctx.parameters["invalidTxt"] = __("codeDigitsError");

  return <><MockStore>{story()}</MockStore>
    <AddListener />
  </>
}


PhoneVerificationSB.decorators = [toastProviderDecorator, verificationDecorator]

/*

export const EmailVerificationSBInteractive = Template.bind({});


// copy args...
EmailVerificationSBInteractive.args = { ...EmailVerificationSB.args }

EmailVerificationSBInteractive.decorators = [toastProviderDecorator, verificationDecorator]

EmailVerificationSBInteractive.storyName = "Interactive";

const PlayTest = async ({ args, canvasElement, parameters }) => {

  console.log("ARGS ", args, parameters);

  const canvas = within(canvasElement);
  const input = canvas.getByTestId('verificationCode');
  console.log("INP ", input);
  const wrongCode = '1234a';
  const goodCode = '123456';

  await userEvent.click(canvas.getByRole("back-click"));
  await expect(args.backClick).toHaveBeenCalled();

  const verifyButton = canvas.getByRole("verify-click");

  // should be still disabled...
  expect(verifyButton.disabled).toBe(true)

  await userEvent.type(input, wrongCode);
  await waitFor(async () => {
    await expect(screen.getByText(parameters.invalidTxt)).toBeInTheDocument();
  });

  //await userEvent.click(canvas.getByRole("verify-click"));
  // should be disabled...
  //await expect(args.verifyClick).not.toHaveBeenCalled();

}


EmailVerificationSBInteractive.play = PlayTest;



// to mockup HOC-component which calls EmailVerification
const ActiveTemplate = args => {

  // this doesn't work in decorator level ... why toast works?
  const { resendCode, verifyCode } = useStore((state) => ({ verifyCode: state.verifyCode, resendCode: state.resendCode }),
    shallow
  );

  const { __ } = useTranslate();
  const alerts = useToast();

  //console.log("STORE", checkCognitoAttributeQuery);

  args.resendClick = () => {
    const actionHandler = action();
    actionHandler("Resend clicked", "email");
    resendCode("email").then(res => {
      alerts.info(__("emailVerificatioSent"), {});
    });
  }

  args.verifyClick = (code) => {
    const actionHandler = action();
    actionHandler("Verify clicked", code);
    verifyCode(code).then(res => {
      console.log("Verified...", res);
    })

  }
  //console.log("ACIVE MOCKUP", args);


  return <>
    <EmailVerification {...args} />
    <AddListener />
  </>
};

export const EmailVerificationSBActive = ActiveTemplate.bind({});



const activeMockupDecorator = (story, ctx) => {

  //console.log("CTX ",ctx.parameters)
  const { __ } = useTranslate();
  ctx.parameters["invalidTxt"] = __("codeDigitsError");

  return <><MockStore ><div>{story()}</div></MockStore></>
}

// copy args...
EmailVerificationSBActive.args = { ...EmailVerificationSB.args }

EmailVerificationSBActive.decorators = [toastProviderDecorator, activeMockupDecorator]

EmailVerificationSBActive.storyName = "Active";

*/