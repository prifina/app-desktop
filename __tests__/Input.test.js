import React from "react";
import renderer from "react-test-renderer";
import {input} from "../stories/Input.stories";
import Input from "../src/components/Input";
import { ThemeProvider } from "@blend-ui/core";

test("Input renders correctly", () => {
    const tree = renderer.create(<ThemeProvider><input type="text" /></ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Disabled renders correctly", () => {
    const tree = renderer.create(
        <ThemeProvider>
            <input disabled type="text" />
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Input with placeholder renders correctly", () => {
    const tree = renderer.create(
        <ThemeProvider>
            <Input placeholder={"Enter value here"} />
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Input Styled no border renders correctly", () => {
    const tree = renderer.create(
        <ThemeProvider>
            <Input placeholder={"Enter value here"} borders={0} p={0} />
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});