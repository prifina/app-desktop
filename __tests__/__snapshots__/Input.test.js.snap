// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disabled renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1w52cfi-0 fESRGN"
>
  <input
    disabled={true}
    type="text"
  />
</div>
`;

exports[`Input Styled no border renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1w52cfi-0 fESRGN"
>
  <input
    className="sc-bdfBwQ eJQRUD"
    placeholder="Enter value here"
    type="text"
    width="100%"
  />
</div>
`;

exports[`Input renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1w52cfi-0 fESRGN"
>
  <input
    type="text"
  />
</div>
`;

exports[`Input with placeholder renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1w52cfi-0 fESRGN"
>
  <input
    className="sc-bdfBwQ jHYwjk"
    placeholder="Enter value here"
    type="text"
    width="100%"
  />
</div>
`;
